File: my-folder/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numTracker = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int complement = target - nums[i];
            int complementIndex = numTracker.getOrDefault(complement, -1);
            if (complementIndex == -1)
                numTracker.put(nums[i], i);
            else
                return new int[]{complementIndex, i};
        }
        return new int[]{-1, -1};
    }
}

File: my-folder/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> solution;
        unordered_map<int, int> targets;
        for (int index = 0; index < nums.size(); ++index) {
            targets[target - nums[index]] = index;
        }

        for (int index = 0; index < nums.size(); ++index) {
            if (targets.contains(nums[index])) {
                if (index != targets[nums[index]]) {
                    solution.push_back(index);
                    solution.push_back(targets[nums[index]]);
                    return solution;
                }
            }
        }

        return solution;
    }
};
